### DISCUSSION TITLE
How to make @root_validator in Pydantic return model class object rather than values in FastAPI view function?

### DISCUSSION THREAD

**Discussion by raghavsikaria** [fastapi/fastapi#4563] [Questions] [answered]
I am currently using a root_validator in my FastAPI project using Pydantic like this:

```python
class User(BaseModel):
    id: Optional[int]
    name: Optional[str]

    @root_validator
    def validate(cls, values):
        if not values.get("id") and not values.get("name"):
            raise ValueError("It's an error")
        return values
```

The issue is, when I query the request body in FastAPI, because of `return values`, instead of the request body being of type `class User`, it is just a simple python dictionary. How do I get the same object of type `class User`?

So initially, my request body would like this when I printed it: `id=0 name='string'` (which is how I want it) (and when I print the `type()` of it, it shows: `<class 'User'>`)

Here is what it looks like with `return values`: `{"id":0, "name"="string"}`

I have tried making it as just `return cls`, but this is what it looks like when I print it: `<class 'User'>` (and when I print the `type()` of it, it shows: `<class 'pydantic.main.ModelMetaclass'>`)

How to get my solution?

**Comment by frankie567** [comment, marked as answer]
I think the problem here is that you called your method `validate`, which is a base method of `BaseModel`. Try to rename it for something else, like `validate_all_filled` and it should work as expected.

**Reply by raghavsikaria** [reply]
Hey, I tried this approach and the only change I made in code was this: `def validate_all_fields(cls, values):` and it threw me an error:

```text
Model values must be a dict; you may not have returned a dictionary from a root validator
```

**Reply by frankie567** [reply]
That's weird. Could you check the type and the actual content of the `values` variable inside your validator?

**Reply by raghavsikaria** [reply]
My bad! My code had `return cls` because of which this error came. I corrected it and now it works like a charm.
Thanks, @frankie567 !!!
